app/jinja_filters.py

```diff
@@ -434,10 +434,25 @@ def __init__(
         answer: SelectFieldBase._Option,
         item_title: str,
         edit_link_text: str,
-        edit_link_aria_label: str,
+        item_name: str | None = None,
     ) -> None:
+        """
+        On post submission summary the change link text is not supposed to be created as there are no change links on the page,
+        we still create SummaryAction class instance even if it's not used (the code is shared and executed by all summaries).
+        Instead of passing an extra argument we check edit_link_text variable contents to find out if we are on post submission or other summary.
+        Also, depending on (list) item_name contents different text is created for list item related answer change link or normal answer change link.
+        """
         self.text = edit_link_text
-        self.visuallyHiddenText = edit_link_aria_label + " " + item_title
+        self.visuallyHiddenText = ""
+        if item_name and edit_link_text:
```

Comment:
There had to be some confusion around the changes at the time of adding them, possibly some other code changes that were removed affected how the logic worked but now these can be improved as per the comment above. Thanks.

Reply:
We still visit this code when generating post submission summary page so we need to check the edit link text since it can be empty, I've added some description in the docstring.

Reply:
Ok, I've implemented the fix you've proposed for the bug discovered, plus the simplification you've suggested above. Thanks.